Code/CryMP/Client/Advertising.cpp

```diff
@@ -0,0 +1,592 @@
+#include "Advertising.h"
+
+#include "CryCommon/CrySystem/ISystem.h"
+#include "CryCommon/CrySystem/IConsole.h"
+#include "CryCommon/CryRenderer/IRenderer.h"
+#include "CryCommon/CryRenderer/IRenderAuxGeom.h"
+
+#include "CryCommon/CryAction/IActorSystem.h"
+#include "CryCommon/CryAction/IVehicleSystem.h"
+#include "CryCommon/CryAction/IMovementController.h"
+#include "CryCommon/CrySystem/ICryPak.h"
+
+#include "CryGame/Actors/Actor.h"
+#include "CryGame/Game.h"
+#include "CryGame/GameCVars.h"
+#include "CryGame/Actors/Player/Player.h"
+#include "CryGame/GameRules.h"
+
+#include "Client.h"
+#include "FileCache.h"
+
+#include "Library/StringTools.h"
+#include "nlohmann/json.hpp"
+
+#include <set>
+#include <string>
+#include <random>
+
+#define AD_CYCLE_MAIN 0
+#define AD_CYCLE_VIEW_CHECK 1
+#define AD_CYCLE_ERROR 2
+
+#define ADVERTISING_SERVER_TOKEN 3000
+
+#ifndef ADS_ENABLED
+#define ADS_ENABLED true
+#endif
+
+#ifndef ADS_SERVER
+#define ADS_SERVER "https://crymp.org/adserve/api/v1/public/ads"
+#endif
+
+struct SAdCue {
+	Vec3 pos;
+	Vec3 entityDir;
+	bool visible;
+};
+
+const std::vector<std::string> knownAdLayersOrdered = {
+	//"auto_defense",
+	"usmilbaseb", "usmilbase",
+	"rynsville", "newfield",
+	"bordercontrol", "bordercheck",
+	"forward", "right" , "left",
+	"sign", "research",
+};
+
+static std::set<std::string> knownAdLayers;
+static std::vector<SAdCue> cues;
+
+static std::random_device rd;
+static std::mt19937 rng(rd());
```

Comment:
There's already one global seeded random generator in [CrySystem/RandomGenerator.cpp](https://github.com/crymp-net/client-server/blob/master/Code/CrySystem/RandomGenerator.cpp).

Reply:
`RandomGenerator::GenerateUInt32` -> [Min, Max]
`RandomGenerator::GenerateIndex` -> [Min, Max)